src/components/gantry/client.ts

```diff
@@ -0,0 +1,144 @@
+import { Struct } from 'google-protobuf/google/protobuf/struct_pb';
+import type { RobotClient } from '../../robot';
+import pb from '../../gen/component/gantry/v1/gantry_pb';
+import { GantryServiceClient } from '../../gen/component/gantry/v1/gantry_pb_service';
+import type { Options, StructType } from '../../types';
+import { doCommandFromClient, promisify } from '../../utils';
+import type { Gantry } from './gantry';
+
+/**
+ * A gRPC-web client for the Gantry component.
+ *
+ * @group Clients
+ */
+export class GantryClient implements Gantry {
+  private client: GantryServiceClient;
+  private readonly name: string;
+  private readonly options: Options;
+
+  constructor(client: RobotClient, name: string, options: Options = {}) {
+    this.client = client.createServiceClient(GantryServiceClient);
+    this.name = name;
+    this.options = options;
+  }
+
+  private get GantryService() {
+    return this.client;
+  }
+
+  async getPosition(extra = {}) {
+    const gantryService = this.GantryService;
+    const request = new pb.GetPositionRequest();
+    request.setName(this.name);
+    request.setExtra(Struct.fromJavaScript(extra));
+
+    this.options.requestLogger?.(request);
+
+    const response = await promisify<
+      pb.GetPositionRequest,
+      pb.GetPositionResponse
+    >(gantryService.getPosition.bind(gantryService), request);
+
+    const result = response.getPositionsMmList();
+    if (!result) {
+      throw new Error('no positions');
+    }
+    return result;
```

Comment:
linter says this will never be falsy so we can return without guarding:

```suggestion
    return response.getPositionsMmList();
```

Reply:
@mcous - should we make `@typescript-eslint/no-unnecessary-condition` an error instead of a warning?